pygeoapi/api.py

```diff
@@ -3435,19 +3430,17 @@ def execute_process(self, request: Union[APIRequest, Any],
         url = f"{self.base_url}/jobs/{job_id}"
 
         headers['Location'] = url
-
-        is_async = data.get('mode', 'auto') == 'async'
-        if is_async:
-            LOGGER.debug('Asynchronous request mode detected')
-
-        if is_async and not self.manager.is_async:
-            LOGGER.debug('async manager not configured/enabled')
-            is_async = False
-
+        try:
+            execution_mode = RequestedProcessExecutionMode(
+                request.headers.get("Prefer"))
+        except ValueError:
+            execution_mode = None
```

Comment:
requested execution mode is now gotten from the `Prefer` HTTP header, if it exists